sdk/src/lib.rs

```diff
@@ -115,6 +117,7 @@ pub struct Worker {
     common: CommonWorker,
     workflow_half: WorkflowHalf,
     activity_half: ActivityHalf,
+    app_data: Option<AppData>,
```

Comment:
Docstring stating why this is optional and how it gets used that way would be a useful reminder